### PR TITLE
added no_data slot to TTable

### PR DESCRIPTION
**Pull request by msorber**
To test check out the add_issues_details_no_data branch, in the issues details page select filters until there are no matching issues and instead of "no data" you should see table contents that match [this](https://www.figma.com/file/IaywW74sLSgYJSYHjOC0ZL/Reporting-Beta-2022?node-id=2882%3A47764).

### CODE REVIEW

File: visualizations/frontend/other/TTable.vue [hunk 1]
```diff
@@ -92,8 +92,10 @@
         "
         class="spanAllColumns center_cell"
       >
-        <div><i class="i-fa-solid i-fa-inbox"></i></div>
-        <div>{{ noDataMessage }}</div>
+        <slot name="no_data" >
+          <div><i class="i-fa-solid i-fa-inbox"></i></div>
```
Comment (sareg0) [inline visualizations/frontend/other/TTable.vue:96]: I don't see any icons in the design. Is this needed?

In addition, I think we might be moving away from fa icons and using more productcl ones, no?

**Review by sareg0 — COMMENTED** [review]
It does not match the design

Screenshot 2022-08-26 at 13 17 00

### Testing setup 

point to this branch in the builder config on https://snyk-insights.topcoatdata.app/a/develop
```
modules:
    - git: https://github.com/topcoat-data/topcoat-public.git
      revision: add_no_data_slot_to_table
```

point to following git branch https://github.com/snyk/snyk-insights-topcoat/tree/add_issues_details_no_data

File: visualizations/frontend/other/TTable.vue (hunk 1 above)
Comment (msorber) [reply visualizations/frontend/other/TTable.vue:96]: The icon is from the [pre-snyk table design](https://docs.google.com/spreadsheets/d/1854rrORMna_BKL7u73I140G9rtrivtcKVNi6EvMSXjs/edit#gid=0). 

Yeah I think we are moving away from fa icons, should probably be its own pitch to go through all the topcoat-public components and replace icons/remove fa dependency.

**Comment by msorber** [comment]
That is odd, I see this with the above setup:

Screen Shot 2022-08-26 at 10 02 26 AM

**Comment by sareg0** [comment]
Okay, I had to rebuild the modules 

Screenshot 2022-08-26 at 17-57-31 Issues Detail

File: visualizations/frontend/other/TTable.vue (hunk 1 above)
Comment (sareg0) [reply visualizations/frontend/other/TTable.vue:96]: I don't see the icons, do you?

**Review by sareg0 — APPROVED** [review]
Do I understand correctly that you are providing a slot for it, because you want different messages per usage of the table?

https://github.com/snyk/snyk-insights-topcoat/pull/17

**Comment by msorber** [comment]
> 

Yep! And the props are not flexible enough to support the above design, hence the need for a slot.

**Pull request merged**